webapp/src/dogma/features/metadata/DeleteMember.tsx

```diff
@@ -19,36 +19,60 @@ import {
 } from '@reduxjs/toolkit/dist/query';
 import { MutationTrigger } from '@reduxjs/toolkit/dist/query/react/buildHooks';
 import { createMessage } from 'dogma/features/message/messageSlice';
-import { DeleteRepoMemberDto } from 'dogma/features/repo/DeleteRepoMemberDto';
+import { DeleteMemberDto } from 'dogma/features/metadata/DeleteMemberDto';
+import { DeleteUserPermissionDto } from 'dogma/features/repo/permissions/DeleteUserPermissionDto';
 import ErrorHandler from 'dogma/features/services/ErrorHandler';
 import { useAppDispatch } from 'dogma/store';
 import { MdDelete } from 'react-icons/md';
 
 export const DeleteMember = ({
   projectName,
+  repoName,
   id,
   deleteMember,
   isLoading,
 }: {
   projectName: string;
+  repoName?: string;
   id: string;
-  deleteMember: MutationTrigger<
-    MutationDefinition<
-      DeleteRepoMemberDto,
-      BaseQueryFn<string | FetchArgs, unknown, FetchBaseQueryError, Record<string, never>, FetchBaseQueryMeta>,
-      'Metadata',
-      void,
-      'api'
-    >
-  >;
+  deleteMember:
+    | MutationTrigger<
+        MutationDefinition<
+          DeleteMemberDto,
+          BaseQueryFn<
+            string | FetchArgs,
+            unknown,
+            FetchBaseQueryError,
+            Record<string, never>,
+            FetchBaseQueryMeta
+          >,
+          'Metadata',
+          void,
+          'api'
+        >
+      >
+    | MutationTrigger<
+        MutationDefinition<
+          DeleteUserPermissionDto,
+          BaseQueryFn<
+            string | FetchArgs,
+            unknown,
+            FetchBaseQueryError,
+            Record<string, never>,
+            FetchBaseQueryMeta
+          >,
+          'Metadata',
+          void,
+          'api'
+        >
+      >;
```

Comment:
Since the type is too complex to understand, should we pass the function as a lambda?
```suggestion
  deleteMember: (projectName: string, repoName: string, id: string) => Promise<void>;
```

The caller may use the changed API like:
```java
<DeleteMember
  projectName={projectName}
  id={info.getValue()}
  deleteMember={(projectName, repoName, id) => deleteMember({ projectName, repoName, id }).unwrap()}
  isLoading={isLoading}
/>
```